cloudslang-compiler/src/test/java/io/cloudslang/lang/compiler/caching/CachedPrecompileServiceImplTest.java

```diff
@@ -0,0 +1,75 @@
+/*******************************************************************************
+ * (c) Copyright 2016 Hewlett-Packard Development Company, L.P.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License v2.0 which accompany this distribution.
+ *
+ * The Apache License is available at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *******************************************************************************/
+package io.cloudslang.lang.compiler.caching;
+
+import io.cloudslang.lang.compiler.SlangSource;
+import io.cloudslang.lang.compiler.modeller.model.Executable;
+import io.cloudslang.lang.compiler.modeller.result.ExecutableModellingResult;
+import java.util.ArrayList;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+
+/**
+ * @author Bonczidai Levente
+ * @since 11/13/2016
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = CachedPrecompileServiceImplTest.Config.class)
+public class CachedPrecompileServiceImplTest {
```

Comment:
hmm, you may be right, but I'd like to work here with a real context, testing `@PostConstruct` and a real `Cache` object